tableauserverclient/server/endpoint/schedules_endpoint.py

```diff
@@ -58,3 +66,28 @@ def create(self, schedule_item):
         new_schedule = ScheduleItem.from_response(server_response.content, self.parent_srv.namespace)[0]
         logger.info("Created new schedule (ID: {})".format(new_schedule.id))
         return new_schedule
+
+    @api(version="2.8")
+    def add_to_schedule(self, schedule_id, workbook=None, datasource=None):
+
+        def add_to(resource, type_, req_factory):
+            id_ = resource.id
+            url = "{0}/{1}/{2}s".format(self.siteurl, schedule_id, type_)
+            add_req = req_factory(id_)
+            response = self.put_request(url, add_req)
+            if response.status_code < 200 or response.status_code >= 300:
+                return AddResponse(result=False,
+                                   error="Status {}: {}".format(response.status_code, response.reason))
+            logger.info("Added {} to {} to schedule {}".format(type_, id_, schedule_id))
+            return OK
+
+        items = []
+
+        if workbook is not None:
+            items.append((workbook, "workbook", RequestFactory.Schedule.add_workbook_req))
+        if datasource is not None:
+            items.append((datasource, "datasource", RequestFactory.Schedule.add_datasource_req))
+
+        results = (add_to(*x) for x in items)
+        # list() is needed for python 3.x compatibility
+        return list(filter(lambda x: not x.result, results))
```

Comment:
I thought harder and rewrote me question: Why filter out the 'bad' responses? How would the user know?

Reply:
👍